File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Ln.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class Ln extends IFunctionPlugin {
	
	@Override
	public String getName()
	{
		return "ln";
	}

	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		double dVal = input.doubleValue();
		if (dVal <= 0 || dVal == Double.NaN)
			throw new FormulaException("ln only accepts positive numbers.");
		
		return new BigDecimal(Math.log(dVal));
	}

}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Ln.scala
package uk.ac.uea.mathsthing.plugins.functions

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin
import java.math.BigDecimal
import uk.ac.uea.mathsthing.Functions
import uk.ac.uea.mathsthing.util.FormulaException

class Ln extends IFunctionPlugin {

  def getName : String = "ln"
    
  def function(input: BigDecimal) : BigDecimal = {
    
    if (input.doubleValue() == Double.NaN || input.doubleValue() <= 0)
      throw new FormulaException("ln only accepts positive numbers.");
    
    val result:BigDecimal = new BigDecimal(Math.log(input.doubleValue))
    result
  }
}